synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -283,8 +283,13 @@ async def action_for_event_by_user(
                 relation.parent_id,
                 itertools.chain(*(r.rules() for r in rules_by_user.values())),
             )
+            # Recursively attempt to find the thread this event relates to.
             if relation.rel_type == RelationTypes.THREAD:
                 thread_id = relation.parent_id
+            else:
+                # Since the event has not yet been persisted we check whether
+                # the parent is parent of a thread.
```

Comment:
```suggestion
                # the parent is child of a thread.
```
?

Reply:
I think it is meant to "we check whether the parent is *part* of a thread."